augur/filter.py

```diff
@@ -1735,6 +1735,10 @@ def numeric_date(date):
     except (ValueError, isodate.ISO8601Error):
         pass
 
+    # This currently doesn't get exposed since argparse raises a SystemExit on invalid arguments.
+    # TODO: find a way to provide better errors for invalid dates.
+    raise ValueError(f"Unable to determine date from {date}.")
```

Comment:
Error messages passed to `argparse.ArgumentTypeError` replace the generic type error and gets surfaced to the user. ( I forget where I read this from...maybe an old stackoverflow?)

Reply:
Oh cool! This fixes from a functional perspective.

Console output with `raise ValueError`:

```
augur filter: error: argument --max-date: invalid numeric_date value: '3000Y'
```

With `raise argparse.ArgumentTypeError`:

```
augur filter: error: argument --max-date: Unable to determine date from 3000Y.
```

However, both surface a `SystemExit` without the actual message in the pytest. Maybe a cram test is more suitable here.

Reply:
Ah, right the pytest. This should be in the SystemExit context, see https://stackoverflow.com/questions/49323496/check-argparse-argumenttypeerror

Reply:
Ohh the message is all the way down in `ExceptionInfo.value.__context__.message`.

Pushed an update: a2ac8e5a